### DISCUSSION TITLE
Focus Fighting

### DISCUSSION THREAD

**Discussion by elliotwoodcinch** [focus-trap/focus-trap#609] [Q&A]
Hi guys 👋  we're having a problem where users become stuck in a focus fighting trap when multiple focus traps are active. Please can you suggest a way around this?
_Eg. a slideout side bar has a focus trap and then an expiry modal opens?_ 
The user is then stuck and can't close either. Should the older modals automatically pause and only leave the latest one active?
Thanks a lot.
Elliot

**Comment by far-fetched** [comment]
Hmm I think if `expiry modal` has its own `focus-trap` this trap would be active, it means that previous trap (from sidebar) is paused. This is how it works. So IMO user should be allowed to do tabbing in new modal. Are you sure, that are you handling some overlays correctly ? Maybe your sidebar has some overlay which is `over` modal and user cannot click on any element which is located in modal ?
Would be wonderful if you provide some simple example.

**Reply by stefcameron** [reply]
This is the correct behavior: When a new focus-trap is activated, the existing one should get paused automatically. And when the second one gets deactivated, the first one one should automatically unpause.
